internal/config/flags.go

```diff
@@ -0,0 +1,154 @@
+/**
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package config
+
+import (
+	"reflect"
+
+	"github.com/NVIDIA/nvidia-container-toolkit/internal/platform-support/tegra/csv"
+
+	cli "github.com/urfave/cli/v2"
+)
+
+// prt returns a reference to whatever type is passed into it
+func ptr[T any](x T) *T {
+	return &x
+}
+
+// ResolveCDIListConfig resolves the config struct for the CDI list subcommand in-place.
+// It sets cfg.cdiSpecDirs using CLI > config > default priority.
+// Accepts *list.config as the third argument.
+func ResolveCDIListConfig(ctx *cli.Context, config *Config, cfg interface{}) {
+	// Use switch statement for type assertion
+	switch v := cfg.(type) {
+	case *struct{ cdiSpecDirs cli.StringSlice }:
+		var dirs []string
+		switch {
+		case ctx.IsSet("spec-dir"):
```

Comment:
In general, I think this is the wrong place to specify this. The flags associated with a config option should not leak to this internal package.

Reply:
As discussed, I am now exploring migrating to Urfave v3 and trying to leverage it for this functionality, will update the PR shortly